packages/core/src/amazonqFeatureDev/util/files.ts

```diff
@@ -102,29 +104,43 @@ export async function collectFiles(
         const files = respectGitIgnore ? await filterOutGitignoredFiles(rootPath, allFiles) : allFiles
 
         for (const file of files) {
-            try {
-                const fileContent = await SystemUtilities.readFile(file, new TextDecoder('utf8', { fatal: true }))
-                const relativePath = getWorkspaceRelativePath(file.fsPath, { workspaceFolders })
+            const { relativePath, fileContent } = await readFile(file, workspaceFolders)
```

Comment:
That call is already reading the full file content into memory.

If the file is 1TB long, will it work?
Can we move the `const fileStat = await vscode.workspace.fs.stat(file)` branch higher?